ci/run_ci.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+#set -eu
+
+# ==============================================================================
+usage() {
+  set +x
+  echo
+  echo "Usage: $0 -d <directory> -o <output> -h"
+  echo
+  echo "  -d  Run build and ctest for clone in <directory>"
+  echo "  -o  Path to output message detailing results of CI tests"
+  echo "  -h  display this message and quit"
+  echo
+  exit 1
+}
+
+# ==============================================================================
+while getopts "d:o:h" opt; do
+  case $opt in
+    d)
+      repodir=$OPTARG
+      ;;
+    o)
+      outfile=$OPTARG
+      ;;
+    h|\?|:)
+      usage
+      ;;
+  esac
+done
+
+# ==============================================================================
+# start output file
+echo "Automated Pull Request Testing Results:" > $outfile
+echo "Machine: ${TARGET}" >> $outfile
+echo '```' >> $outfile
+echo "Start: $(date) on $(hostname)" >> $outfile
+echo "---------------------------------------------------" >> $outfile
+# ==============================================================================
+# run build script
+cd $repodir
+module purge
+./build.sh -t $TARGET &>> log.build
+build_status=$?
+if [ $build_status -eq 0 ]; then
+  echo "Build:                                 *SUCCESS*" >> $outfile
+  echo "Build: Completed at $(date)" >> $outfile
+else
+  echo "Build:                                  *FAILED*" >> $outfile
+  echo "Build: Failed at $(date)" >> $outfile
+  echo "Build: see output at $repodir/log.build" >> $outfile
+  echo '```' >> $outfile
+  exit $build_status
+fi
+# ==============================================================================
+# run ctests
+cd $repodir/build
```

Comment:
@CoryMartin-NOAA , why not limit the testing to this repo only and ignore the test status of the jedi components? It might sounds silly right now, but in the next few pr's we will have specific tests for the "prep" "run" and "post" steps, which will be making use of the various jedi components. I think it's a good enough "canary".
I vote for ``ctest -R gdasapp`` or something of that effect.
Or maybe 2 reports? One for the jedi components and one for the gdasapp tests?

Reply:
Nope, we might as well do the build now. We're going to start adding tests using the jedi apps soon enough.
That's currently what I'm working on, just empty stubs right now but I'm hoping to have a basic jedi/soca var application by the end of the day:
```console
build$ ctest -R gdasapp
Test project /home/gvernier/sandboxes/GDASApp/build
    Start 1448: test_gdasapp_soca_obsdb
1/4 Test #1448: test_gdasapp_soca_obsdb ..........   Passed    0.27 sec
    Start 1449: test_gdasapp_soca_ana_prep
2/4 Test #1449: test_gdasapp_soca_ana_prep .......   Passed    0.01 sec
    Start 1450: test_gdasapp_soca_ana_run
3/4 Test #1450: test_gdasapp_soca_ana_run ........   Passed    0.00 sec
    Start 1451: test_gdasapp_soca_ana_post
4/4 Test #1451: test_gdasapp_soca_ana_post .......   Passed    0.00 sec

100% tests passed, 0 tests failed out of 4

Label Time Summary:
gdasapp    =   0.29 sec*proc (4 tests)
script     =   0.29 sec*proc (4 tests)
```

Reply:
We'll need to remember adding `gdasapp` (or something) to the one test we currently have. It's now called ``test_check_yaml_keys`` so something like ``test_gdasapp_check_yaml_keys``, or whatever consensus we decide on.